service/tablet_allocator.cc

```diff
@@ -147,6 +147,51 @@ struct migration_badness {
     bool operator<=>(const migration_badness& other) const = default;
 };
 
+struct colocated_tablet {
+    global_tablet_id left_tablet;
+    global_tablet_id right_tablet;
+
+    auto operator<=>(const colocated_tablet&) const = default;
+};
+
+template <class... Ts>
+struct overloaded : Ts... {
+    using Ts::operator()...;
+};
+
+struct tablet_candidate {
```

Comment:
tablet_merge_candidate?

Reply:
The reason I preferred to name it tablet_candidate (I also thought of balancer_candidate) is because we're using this not only for co-location work, but also for regular balancing. Once wired, it will be what the balancer will be working with, even when the context is not merge.